### PR TITLE
Done list of the homework tasks from the Lesson Podstawy C++ 2 czesc 

### PR DESCRIPTION
**Pull request by mikailromer**
Define NWD and NWW function.
Define generateSequence function.
Define maxOfVector function.
Define addEven function.

### CODE REVIEW

File: module2/homework/addEven/addEven.hpp [hunk 1]
```diff
@@ -2,7 +2,10 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int sum = 0;
+    for (const auto& elem : numbers)
+    {
+        if((elem)%2 == 0) sum+= elem;
```
Comment (piezol) [inline module2/homework/addEven/addEven.hpp:8]:
```suggestion
        if(elem % 2 == 0) {
            sum += elem;
        }
```
Spacing

File: module2/homework/generateSequence/generateSequence.hpp [hunk 2]
```diff
@@ -2,6 +2,14 @@
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    if(count <= 0) return std::vector<int>{};
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:5]:
```suggestion
    if(count <= 0) {
        return std::vector<int>{};
    }
```

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 3]
```diff
@@ -3,6 +3,10 @@
 #include <limits>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    auto max = vec[0];
+    for(const auto& elem : vec)
+    {
+        if(max < elem) max = elem;
```
Comment (piezol) [inline module2/homework/maxOfVector/maxOfVector.hpp:9]:
```suggestion
        if(max < elem) {
            max = elem;
        }
```

File: module2/homework/nwdNww/nwdNww.hpp [hunk 4]
```diff
@@ -1,11 +1,68 @@
...
+    
+    if(rhs == 0 || lhs == 0)
+    {
+        if(rhs == 0 && lhs != 0) return lhs;
+        else if(rhs != 0 && lhs == 0) return rhs;
+        else return 0;
+    }
+
+    if(rhs < 0) rhs = abs(rhs);
+    if(lhs < 0) lhs = abs(lhs);
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:14]:
```suggestion
    lhs = std::abs(lhs);
    rhs = std::abs(rhs);
```

File: module2/homework/nwdNww/nwdNww.hpp [hunk 5]
```diff
@@ -1,11 +1,68 @@
 #pragma once
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:2]: You need to include a file that defines abs (or implement your own).

File: module2/homework/nwdNww/nwdNww.hpp [hunk 6]
```diff
@@ -1,11 +1,68 @@
...
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    int dividend, divider, result = 1;
+    
+    if(rhs == 0 || lhs == 0)
+    {
+        if(rhs == 0 && lhs != 0) return lhs;
+        else if(rhs != 0 && lhs == 0) return rhs;
+        else return 0;
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:10]:
```suggestion
        if(rhs == 0 && lhs != 0) {
            return lhs;
        } else if(rhs != 0 && lhs == 0) {
            return rhs;
        }  
        return 0;

```
That's as formatting goes.
But i've seen even better solution: if either lhs or rhs is equal zero, return lhs + rhs.
No extra if statements.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 7]
```diff
@@ -1,11 +1,68 @@
...
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    int dividend, divider, result = 1;
+
+    if(rhs == 0 || lhs == 0) return 0;
+
+    if(rhs < 0) rhs = abs(rhs);
+    if(lhs < 0) lhs = abs(lhs);
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:45]:
```suggestion
    lhs = std::abs(lhs);
    rhs = std::abs(rhs);
```

File: module2/homework/nwdNww/nwdNww.hpp [hunk 8]
```diff
@@ -1,11 +1,68 @@
...
+    for(int i = dividend; ; i++)
+    {
+        if(i % dividend  == 0 && i % divider == 0)
+        {
+            result = i;
+            break;
+        }
+    }
+
+    return result;
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:67]: Try to find more efficient algorithm for finding LCM. 
Your approach is very slow (it basically checks all the possibilities).

It is fine if your version doesn't return correct result for tricky inputs, but in your case there are inputs for which the algorithm would never end!
Try out 1e8-1, 1e8+1.
I overflows before it reaches LCM.

Even if the algorithm wouldn't require change:
loops without known number of iterations are usually better written using 'while'.
You don't need artificial variable to keep the result, you can just return it inside loop.

File: module2/homework/addEven/addEven.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:8]: space

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**